extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -0,0 +1,254 @@
+    this.document_ = this.win_.document;
+
+    /** @private @const {!Element} */
+    this.documentElement_ = this.document_.documentElement;
+
+    /** @private @const {string} */
+    this.direction_ = this.element.getAttribute('direction');
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = this.getViewport();
+
+    /** @private @const {boolean} */
+    this.hasMask_ = false;
+
+    /** @private @const {boolean} */
+    this.isIosSafari_ = platform.isIos() && platform.isSafari();
+
+    if (this.direction_ != 'left' && this.direction_ != 'right') {
+      const pageDir = this.document_.body.getAttribute('dir') || 'ltr';
+      this.direction_ = (pageDir == 'rtl') ? 'right' : 'left';
+      this.element.setAttribute('direction', this.direction_);
+    }
+
+    if (!this.isExperimentOn_) {
+      dev.warn(TAG, `Experiment ${EXPERIMENT} disabled`);
+      return;
+    }
+    if (!this.checkWhitelist_()) {
+      return;
+    }
+
+    if (this.isIosSafari_) {
+      this.fixIosElasticScrollLeak_();
+    }
+
+    this.adjustPadding_();
+
+    if (this.documentElement_.classList.contains('amp-sidebar-open')) {
+      // Create the mask if the sidebar is rendered in open mode.
+      this.createMask_();
+    }
+
+    this.documentElement_.addEventListener('keydown', event => {
+      // Close sidebar on ESC.
+      if (event.keyCode == 27) {
+        this.close_();
+      }
+    });
+    //TODO (skrish, #2712) Add history support on back button.
+    this.registerAction('activate', this.open_.bind(this));
+    this.registerAction('toggle', this.toggle_.bind(this));
+    this.registerAction('open', this.open_.bind(this));
+    this.registerAction('close', this.close_.bind(this));
+  }
+
+  /**
+   * @private
+   */
+  adjustPadding_() {
+    const viewerPaddingTop = this.viewport_.getPaddingTop();
+    if (viewerPaddingTop) {
+      // viewerPaddingTop exists when AMP page is rendered inside search
+      // carousel.
+      const div = this.document_.createElement('div');
+      setStyles(div, {
+        'height': viewerPaddingTop + 'px',
+        'width': '100%',
+      });
+      const firstChild = this.element.firstChild;
+      this.element.insertBefore(div, firstChild);
+    }
+    if (this.isIosSafari_) {
+      //Compensate for IOS safari bottom navbar.
+      const div = this.document_.createElement('div');
+      setStyles(div, {
+        'height': IOS_SAFARI_BOTTOMBAR_HEIGHT,
+        'width': '100%',
+      });
+      this.element.appendChild(div);
+    }
+  }
+
+  /**
+   * Toggles the open/close state of the sidebar.
+   * @private
+   */
+  toggle_() {
+    if (this.documentElement_.classList.contains('amp-sidebar-open')) {
+      this.close_();
+    } else {
+      this.open_();
+    }
+  }
+
+  /**
+   * Reveals the sidebar.
+   * @private
+   */
+  open_() {
+    this.mutateElement(() => {
+      this.getViewport().addToFixedLayer(this.element);
+      this.createMask_();
+      this.viewport_.disableTouchZoom();
```

Comment:
move above the `mutateElement` - we would want to reset zooming before UI mutations start.